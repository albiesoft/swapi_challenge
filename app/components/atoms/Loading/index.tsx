import { FC, useState } from 'react';
import { Spin } from 'antd';
import Router from 'next/router';
import { LoadingOutlined } from '@ant-design/icons';

const antIcon = <LoadingOutlined style={{ fontSize: 24 }} spin />;

const Loading: FC = ({ children }) => {
  const [loanding, setLoanding] = useState(false);
  Router.events.on('routeChangeStart', () => setLoanding(true));
  Router.events.on('routeChangeComplete', () => setLoanding(false));
  Router.events.on('routeChangeError', () => setLoanding(false));
  
  if (loanding) {
    return (
      <Spin indicator={antIcon} tip="Loading...">
        {children}
      </Spin>
    );
  }

  return (
    <>
      {children}
    </>
  );
}

export default Loading;
