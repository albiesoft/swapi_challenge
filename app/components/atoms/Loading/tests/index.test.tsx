import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Loading from '..';

describe('<Loading />', () => {
  test('It renders correctly initially', () => {
    const target: ShallowWrapper = shallow(<Loading />);
    expect(target).toMatchSnapshot();
  });
});
