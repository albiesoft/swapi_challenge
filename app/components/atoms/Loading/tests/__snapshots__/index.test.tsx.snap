// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> It renders correctly initially 1`] = `<Fragment />`;
